### PR TITLE
CLN: Isort core/*.py and core/indexes/

### PR DESCRIPTION
**Pull request by alimcmaster1**
- [x] xref #23334
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: pandas/core/apply.py [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-    is_dict_like,
-    is_list_like,
-    is_sequence)
 from pandas.util._decorators import cache_readonly
 
+from pandas.core.dtypes.common import (
+    is_dict_like, is_extension_type, is_list_like, is_sequence)
+from pandas.core.dtypes.generic import ABCSeries
+
+from pandas import compat
```
Comment (jbrockmendel) [inline pandas/core/apply.py]: Can you make this “import pandas.compat as compat”?  That will group it with the appropriate level of dependencies.  Same in other files if it comes up.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -1,64 +1,54 @@
...
+    Index, InvalidIndexError, MultiIndex, RangeIndex, ensure_index)
 from pandas.core.indexes.datetimes import DatetimeIndex
-from pandas.core.indexes.period import PeriodIndex, Period
+from pandas.core.indexes.period import Period, PeriodIndex
+import pandas.core.indexing as indexing
 from pandas.core.internals import BlockManager
-import pandas.core.algorithms as algos
-import pandas.core.common as com
 import pandas.core.missing as missing
+import pandas.core.nanops as nanops
```
Comment (jbrockmendel) [inline pandas/core/generic.py]: Can you put this in the same line as “from pandas.core import config”? Dittto “missing” one line up

File: pandas/core/indexes/api.py [hunk 3]
```diff
@@ -1,23 +1,21 @@
...
-                                      _new_Index,
-                                      ensure_index,
-                                      ensure_index_from_sequences,
-                                      InvalidIndexError)  # noqa
+from pandas._libs import NaT, lib
+
+import pandas.core.common as com
+from pandas.core.indexes.base import (
+    Index, _new_Index, ensure_index, ensure_index_from_sequences)
+from pandas.core.indexes.base import InvalidIndexError  # noqa
```
Comment (jbrockmendel) [inline pandas/core/indexes/api.py]: Her and elsewhere can you make the “noqa” specific?  I think it’s “noqa: F401”

Comment (alimcmaster1) [reply pandas/core/indexes/api.py]: Sure makes sense! Thanks

File: pandas/core/apply.py (hunk 1 above)
Comment (alimcmaster1) [reply pandas/core/apply.py]: Thanks for this ~11 files in this PR. ( Will make this change in my other PR too )

File: pandas/core/generic.py (hunk 2 above)
Comment (alimcmaster1) [reply pandas/core/generic.py]: Updated, thanks

**Comment by alimcmaster1** [comment]
@jbrockmendel thanks for the review updated as per your comments

**Comment by jreback** [comment]
can you merge master. ping on green.

File: setup.cfg [hunk 4]
```diff
@@ -111,33 +111,8 @@ force_grid_wrap=0
...
 force_sort_within_sections=True
 skip=
-    pandas/core/ops.py,
-    pandas/core/categorical.py,
     pandas/core/api.py,
-    pandas/core/indexing.py,
-    pandas/core/apply.py,
-    pandas/core/generic.py,
-    pandas/core/sorting.py,
     pandas/core/frame.py,
```
Comment (alimcmaster1) [inline setup.cfg]: Sorting frame.py seems to cause a few problems will address this in separate PR

**Comment by alimcmaster1** [comment]
merged master - cc. @jreback

**Comment by jreback** [comment]
thanks @alimcmaster1

**Pull request merged**